core-plugins/src/main/java/co/cask/hydrator/plugin/common/BatchFileFilter.java

```diff
@@ -73,9 +75,13 @@
   @Override
   public boolean accept(Path path) {
     String filePathName = path.toString();
-    //The path filter will first check the directory if a directory is given
-    if (filePathName.equals(pathName) || filePathName.equals(pathName + "/")) {
-      return true;
+    try {
+      FileSystem fileSystem = path.getFileSystem(new Configuration());
```

Comment:
so this will now accept any directory, even directories that cannot match the pattern. That can be significant overhead